### PR TITLE
cherry-pick hazelcast remote distribution test from 9.4.x (#6068)

### PR DESCRIPTION
**Pull request by olamy**
* Add distribution test using a remote hazelcast only client true and false, false is failing with CNFE

Signed-off-by: Olivier Lamy <oliver.lamy@gmail.com>

### CODE REVIEW

File: jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod [hunk 1]
```diff
@@ -14,18 +14,18 @@ sessions
...
 maven://com.hazelcast/hazelcast/${hazelcast.version}|lib/hazelcast/hazelcast-${hazelcast.version}.jar
-maven://com.hazelcast/hazelcast-client/${hazelcast.version}|lib/hazelcast/hazelcast-client-${hazelcast.version}.jar
 
 [xml]
 etc/sessions/hazelcast/remote.xml
 
 [lib]
 lib/jetty-hazelcast-${jetty.version}.jar
 lib/hazelcast/hazelcast-${hazelcast.version}.jar
-lib/hazelcast/hazelcast-client-${hazelcast.version}.jar
```
Comment (olamy) [inline jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod:25]: interesting to find a bug we have here with this test ;)

**Comment by olamy** [comment]
@janbartel nudge :)

File: jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod [hunk 2]
```diff
@@ -14,18 +14,18 @@ sessions
...
 
 [ini]
 hazelcast.version?=4.1
+jetty.webapp.addSystemClasses+=,org.eclipse.jetty.hazelcast.
+jetty.webapp.addServerClasses+=,-org.eclipse.jetty.hazelcast.
```
Comment (janbartel) [inline jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod]: I'm not sure this is totally necessary as org.eclipse.jetty package is a server class by default, isn't it?

**Review by janbartel — DISMISSED** [review]
I'm a little cautious about making these system classes, just because we haven't made any of the other session integrations system classes, but we _have_ made most of their dependencies server classes.

File: jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod (hunk 2 above)
Comment (olamy) [reply jetty-hazelcast/src/main/config/modules/session-store-hazelcast-remote.mod]: ah yup good catch. not needed for 10.

**Comment by olamy** [comment]
@janbartel you need to approve again (sorry for that but that's the github process...)

**Review by janbartel — APPROVED** [review]

**Pull request merged**